### PR TITLE
Add cluster monitoring options and system image

### PR DESCRIPTION
**Pull request by galal-hussein**

### CODE REVIEW

File: apis/management.cattle.io/v3/rke_types.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ type RancherKubernetesEngineConfig struct {
 	AddonJobTimeout int `yaml:"addon_job_timeout" json:"addonJobTimeout,omitempty" norman:"default=30"`
 	// Bastion/Jump Host configuration
 	BastionHost BastionHost `yaml:"bastion_host" json:"bastionHost,omitempty"`
+	// Disable Metrics Server addon
+	DisableMetricsServer bool `yaml:"disable_metrics_server" json:"disableMetricsServer,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/rke_types.go]: why bool? can we make it similar to ingress? so it has a provider and a list of options

Comment (galal-hussein) [reply apis/management.cattle.io/v3/rke_types.go]: hmmm, i see, okay will fix

File: apis/management.cattle.io/v3/rke_types.go [hunk 2]
```diff
@@ -552,3 +556,11 @@ type AzureCloudProvider struct {
 // AWSCloudProvider options
 type AWSCloudProvider struct {
 }
+
+// MetricsServerConfig
+type MetricsServerConfig struct {
+	// Disable metrics server
+	Disable bool `yaml:"disable" json:"disable,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/rke_types.go]: can we just use None value to disable it? again, similar to other addons

Comment (alena1108) [reply apis/management.cattle.io/v3/rke_types.go]: and whatever provider is used for the current metric server as an alternative

Comment (galal-hussein) [reply apis/management.cattle.io/v3/rke_types.go]: the problem is that metrics server doesn't have plugins like ingress, but it does contain options

**Comment by alena1108** [comment]
LGTM

**Pull request merged**